### PR TITLE
[FEATURE] Add specVersion 3.1 and builder resource excludes for modules

### PR DESCRIPTION
**Pull request by RandomByte**
Based on https://github.com/SAP/ui5-project/pull/623 but with the component type removed again.

JIRA: CPOUI5FOUNDATION-561

### CODE REVIEW

File: lib/specifications/types/Library.js [hunk 1]
```diff
@@ -107,17 +107,18 @@ class Library extends ComponentProject {
 	}
 
 	/**
-	 *
 	 * Get a resource reader for the sources of the project (excluding any test resources)
+	 * without a virtual base path
```
Comment (flovogt) [inline lib/specifications/types/Library.js:111]:
```suggestion
	 * without a virtual base path.
```

File: lib/specifications/SpecificationVersion.js [hunk 2]
```diff
@@ -263,6 +263,22 @@ for example <code>2.2 - 2.4</code>
 		const comparator = new SpecificationVersion(specVersion);
 		return comparator.neq(testVersion);
 	}
+
+	/**
+	 * Create an array of Specification Versions that match with the provided range. This is mainly used
```
Comment (flovogt) [inline lib/specifications/SpecificationVersion.js:268]:
```suggestion
	 * Creates an array of Specification Versions that match with the provided range. This is mainly used
```

**Review by d3xter666 — DISMISSED** [review]
LGTM

**Review by flovogt — CHANGES_REQUESTED** [review]
Well done. Awesome improvement! Only minor comments

File: test/lib/validation/schema/specVersion/kind/project/module.js [hunk 3]
```diff
@@ -417,4 +418,25 @@ test.after.always((t) => {
...
+			"metadata": {
+				"name": "my-module"
+			},
+			"builder": {
+				"resources": {
+					"excludes": [
+						"/resources/some/project/name/test_results/**",
+						"!/test-resources/some/project/name/demo-app/**"
+					]
+				}
```
Comment (matz3) [inline test/lib/validation/schema/specVersion/kind/project/module.js:436]: What about `settings`? They are now also allowed. Was this intended?

File: lib/validation/schema/specVersion/kind/project/module.json [hunk 4]
```diff
@@ -149,6 +184,18 @@
...
+		"builder-specVersion-3.1": {
+			"type": "object",
+			"additionalProperties": false,
+			"properties": {
+				"resources": {
+					"$ref": "../project.json#/definitions/builder-resources"
+				},
+				"settings": {
+					"$ref": "../project.json#/definitions/builder-settings"
+				}
```
Comment (matz3) [inline lib/validation/schema/specVersion/kind/project/module.json:197]: I think this also makes sense to add, but we never talked about it, right?

**Review by matz3 — CHANGES_REQUESTED** [review]

File: lib/validation/schema/specVersion/kind/project/module.json (hunk 4 above)
Comment (RandomByte) [reply lib/validation/schema/specVersion/kind/project/module.json:197]: Settings were already allowed before. See https://github.com/SAP/ui5-project/pull/639/files#diff-7b0836b8d51420484c25fc8977caba9234b6efc7193d3f0507731f291f8362beR183 and https://github.com/SAP/ui5-project/pull/639/files#diff-a21c72474996eb0b320d874991042a771b23940e75ce1ce9344261143d8bf875R184

Comment (matz3) [reply lib/validation/schema/specVersion/kind/project/module.json:197]: Thanks. I missed that.

**Review by matz3 — APPROVED** [review]

**Review by flovogt — APPROVED** [review]

**Pull request merged**